cupyx/jit/_builtin_funcs.py

```diff
@@ -219,6 +219,34 @@ def call_const(self, env, dtype, size, alignment=None):
         env.locals[name] = var
         return Data(name, _cuda_types.Ptr(ctype))
 
+class LocalMemory(BuiltinFunc):
+
+    def __call__(self, dtype, size, alignment=None):
+        """Allocates local memory and returns it as a 1-D array.
+
+        Args:
+            dtype (dtype):
+                The dtype of the returned array.
+            size (int or tuple):
+                If ``int`` type, the size of static local memory.
+                If ``tuple`` type, the size of a multi-dimensional static local memory.
+                Does not use __extern__ keyword like shared memory does
+            alignment (int or None): Enforce the alignment via __align__(N).
+        """
+        super().__call__()
+
+    def call_const(self, env, dtype, size, alignment=None):
+        name = env.get_fresh_variable_name(prefix='_lmem')
+        ctype = _cuda_typerules.to_ctype(dtype)
+        var = Data(name, _cuda_types.LocalMem(ctype, size, alignment))
+        env.decls[name] = var
+        env.locals[name] = var
+        if(type(size) is int):
```

Comment:
The following code raises `TypeError`.
```py
@jit.rawkernel()
def f():
    lmem = jit.local_memory(numpy.int32, numpy.int64(10))
```

```suggestion
        if not isinstance(size, tuple):
```